File: python-twisted/tests/unit-test.py
## www.pubnub.com - PubNub Real-time push service in the cloud. 
# coding=utf8

## PubNub Real-time Push APIs and Notifications Framework
## Copyright (c) 2010 Stephen Blum
## http://www.pubnub.com/

## -----------------------------------
## PubNub 3.1 Real-time Push Cloud API
## -----------------------------------

import sys
sys.path.append('../')
from Pubnub import Pubnub

publish_key   = len(sys.argv) > 1 and sys.argv[1] or 'demo'
subscribe_key = len(sys.argv) > 2 and sys.argv[2] or 'demo'
secret_key    = len(sys.argv) > 3 and sys.argv[3] or 'demo'
cipher_key    = len(sys.argv) > 4 and sys.argv[4] or ''      ##(Cipher key is Optional)
ssl_on        = len(sys.argv) > 5 and bool(sys.argv[5]) or False

## -----------------------------------------------------------------------
## Initiat Class
## -----------------------------------------------------------------------
pubnub = Pubnub( publish_key, subscribe_key, secret_key, cipher_key, ssl_on )
crazy  = ' ~`!@#$%^&*( 顶顅 Ȓ)+=[]\\{}|;\':",./<>?abcd'

## ---------------------------------------------------------------------------
## Unit Test Function
## ---------------------------------------------------------------------------
def test( trial, name ) :
    if trial :
        print( 'PASS: ' + name )
    else :
        print( 'FAIL: ' + name )

## -----------------------------------------------------------------------
## Time Example
## -----------------------------------------------------------------------
def time_complete(timestamp):
    print(timestamp)

pubnub.time({ 'callback' : time_complete })

## -----------------------------------------------------------------------
## History Example
## -----------------------------------------------------------------------
def history_complete(messages):
    print(messages)

pubnub.history( {
    'channel'  : crazy,
    'limit'    : 10,
    'callback' : history_complete
})

## -----------------------------------------------------------------------
## Publish Example
## -----------------------------------------------------------------------
def publish_complete(info):
    print(info)

pubnub.publish({
    'channel' : crazy,
    'message' :  {'one': 'Hello World! --> ɂ顶@#$%^&*()!', 'two': 'hello2'},
    'callback' : publish_complete
})

## -----------------------------------------------------------------------
## Subscribe Example
## -----------------------------------------------------------------------
def message_received(message):
    print(message)
    print('Disconnecting...')
    pubnub.unsubscribe({ 'channel' : crazy })

    def done() :
        print('final connection, done :)')
        pubnub.unsubscribe({ 'channel' : crazy })
        pubnub.stop()

    def dumpster(message) :
        print('never see this')
        print(message)

    print('reconnecting...')
    pubnub.subscribe({
        'channel'  : crazy,
        'connect'  : done,
        'callback' : dumpster
    })

def connected() :
    pubnub.publish({
        'channel' : crazy,
        'message' : { 'Info' : 'Connected!' }
    })

pubnub.subscribe({
    'channel'  : crazy,
    'connect'  : connected,
    'callback' : message_received
})
## -----------------------------------------------------------------------
## IO Event Loop
## -----------------------------------------------------------------------
pubnub.start()

File: ruby/tests/unit_test.rb
# encoding: utf-8
## www.pubnub.com - PubNub realtime push service in the cloud.
## http://www.pubnub.com/blog/ruby-push-api - Ruby Push API Blog

## PubNub Real Time Push APIs and Notifications Framework
## Copyright (c) 2010 Stephen Blum
## http://www.pubnub.com/

## -----------------------------------
## PubNub 3.1 Real-time Push Cloud API
## -----------------------------------

## including required libraries
require 'rubygems'
require 'pubnub'

## declaring publish_key, subscribe_key, secret_key, cipher_key, message, ssl_on (Cipher key is Optional)
publish_key   = ARGV[0] || 'demo'
subscribe_key = ARGV[1] || 'demo'
secret_key    = ARGV[2] || 'demo'
cipher_key    = ARGV[3] || ''    # (Cipher key is Optional)
ssl_on        = false
channel       = 'hello_world'
message       = 'Hi. (顶顅Ȓ)'

## ---------------------------------------------------------------------------
## Create Pubnub Client API (INITIALIZATION)
## ---------------------------------------------------------------------------
puts('Initializing new Pubnub state')
pubnub = Pubnub.new( publish_key, subscribe_key, secret_key,cipher_key, ssl_on )

## ---------------------------------------------------------------------------
## Unit Test Function
## ---------------------------------------------------------------------------
def test( trial, name )
  trial ? puts('PASS: ' + name) : puts('FAIL: ' + name)
end

## ---------------------------------------------------------------------------
## UUID generation
## ---------------------------------------------------------------------------
uuid = pubnub.UUID()
test( uuid.length > 0, 'PubNub Client API UUID: ' + uuid )

## ---------------------------------------------------------------------------
## PubNub Server Time
## ---------------------------------------------------------------------------
pubnub.time({'callback' => lambda do |message|
     print 'PubNub Server Time: ', message ## print message
  end
})

## ---------------------------------------------------------------------------
## PUBLISH
## ---------------------------------------------------------------------------
pubnub.publish({
  'channel' => channel,
  'message' => message,
  'callback' => lambda do |message|
    puts(message)
  end
})
#test( pubish_success[0] == 1, 'Publish First Message Success' )

## ---------------------------------------------------------------------------
## HISTORY
## ---------------------------------------------------------------------------
pubnub.history({
  'channel' => channel,
  'limit'   => 5,
  'callback' => lambda do |message|
    puts(message)
  end
})
#puts(history)
#test( history.length >= 1, 'Display History' )

## ---------------------------------------------------------------------------
## Subscribe
## ---------------------------------------------------------------------------
puts('Listening for new messages with subscribe() Function')
puts('Press CTRL+C to quit.')
pubnub.subscribe({
  'channel'  => channel,
  'callback' => lambda do |message|
  puts(message) ## print message
  return true   ## keep listening?
  end
})
